contracts/hyphen/LiquidityPool.sol

```diff
@@ -286,6 +282,8 @@ contract LiquidityPool is ReentrancyGuardUpgradeable, Pausable, OwnableUpgradeab
         processedHash[hashSendTransaction] = true;
 
         uint256 amountToTransfer = getAmountToTransfer(initialGas, tokenAddress, amount, tokenGasPrice);
+        liquidityProviders.decreaseCurrentLiquidity(tokenAddress, amountToTransfer);
```

Comment:
Should we not decrease the amount directly, since the (amount - amountToTransfer) will be distributed into transferFee, gasFee and incentiveFee?

Reply:
that's the point, amount - amountToTransfer will already be taken care of in the method getCurrentLiquidity(), lpFee, incentive fee, gasFee are updated in separate places... so to avoid updating currentLiquidity map at multiple places.. i've updated only the effective incoming and outgoing liquidity on contract.